auth_service/auth_adapter/core/auth.py

```
@@ -18,121 +18,207 @@
+            raise ValueError("The subject claim is missing.")
+        for claim in jwt_config.copied_claims:
+            if not claims[claim]:
+                raise ValueError(f"Missing value for {claim} claim.")
+        return claims
+    except Exception as exc:  # pylint:disable=broad-except
+        log.debug("Cannot validate external token: %s", exc)
     return None
 
 
-def sign_and_encode_token(payload: dict[str, Any], key=None) -> Optional[str]:
+def sign_and_encode_token(claims: dict[str, Any], key=None) -> Optional[str]:
     """Encode and sign the given payload as JSON Web Token.
 
     Returns None in case the payload could not be properly signed.
     """
-    if not payload:
+    if not claims:
         return None
     if not key:
-        key = signing_keys.internal_jwk
+        key = jwt_config.internal_jwk
         if not key:
-            log.debug("No internal signing key, cannot sign token")
+            log.debug("No internal signing key, cannot sign token.")
             return None
     try:
-        jws_token = jws.JWS(json.dumps(payload).encode("utf-8"))
-        header = json.dumps({"kid": key.thumbprint()})
-        alg = "ES256" if key["kty"] == "EC" else "RS256"
-        protected = {"alg": alg}
-        jws_token.add_signature(key, alg=alg, protected=protected, header=header)
-        token = jws_token.serialize(compact=True)
-    except (
-        jws.InvalidJWSObject,
-        jws.InvalidJWSOperation,
-        UnicodeEncodeError,
-        ValueError,
-    ) as error:
-        log.error("Error while signing JTW: %s", error)  # pragma: no cover
-        token = None
+        header = {"alg": "ES256" if key["kty"] == "EC" else "RS256", "typ": "JWT"}
+        token = jwt.JWT(header=header, claims=claims)
+        token.make_signed_token(key)
+        return token.serialize(compact=True)
+    except Exception as exc:  # pylint:disable=broad-except
+        log.error("Error while signing JTW: %s", exc)  # pragma: no cover
```

Again, maybe just let it fail.

---

Changed in 2a334db.